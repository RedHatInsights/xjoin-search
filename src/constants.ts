export const IDENTITY_HEADER = 'x-rh-identity';

export const ES_NULL_VALUE = '$$_XJOIN_SEARCH_NULL_VALUE';
