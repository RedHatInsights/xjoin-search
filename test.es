

GET /test.hosts.v1/_mapping

GET /test.hosts/_doc/_search
