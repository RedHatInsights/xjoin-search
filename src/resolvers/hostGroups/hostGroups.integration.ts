import { runQuery, createHeaders } from '../../../test/helpers';
import * as constants from '../../constants';
import createIdentityHeader from '../../middleware/identity/utils';

const GROUP_FILTERS_QUERY = `
    query hostGroups (
        $hostFilter: HostFilter,
        $order_by: HOST_GROUPS_ORDER_BY,
        $order_how: ORDER_DIR,
        $limit: Int,
        $offset: Int) {
        hostGroups (
            hostFilter: $hostFilter,
            order_by: $order_by,
            order_how: $order_how,
            limit: $limit,
            offset: $offset
        )
        {
            meta {
                count,
                total
            }
            data {
                group {
                    id, name, account, org_id, created_on, modified_on
                },
                count
            }
        }
    }
`;

describe('host groups', function () {
    test('basic query', async () => {
        const {data, status} = await runQuery(GROUP_FILTERS_QUERY, {});
        expect(status).toEqual(200);
        expect(data).toMatchSnapshot();
    });

    test('org_id isolation', async () => {
        const headers = {
            [constants.IDENTITY_HEADER]: createIdentityHeader(
                f => f, 'customer', '12345', false)
        };

        const { data, status } = await runQuery(GROUP_FILTERS_QUERY, {}, headers);
        data.hostGroups.data.should.have.length(0);
        data.hostGroups.meta.should.have.property('total', 0);
        data.hostGroups.meta.should.have.property('count', 0);
        expect(status).toEqual(200);
    });

    describe('pagination', function () {
        const headers = createHeaders('hostGroupsTest', 'hostGroupsTest');

        test('limit', async () => {
            const {data, status} = await runQuery(GROUP_FILTERS_QUERY, {
                limit: 2
            }, headers);

            expect(status).toEqual(200);
            data.hostGroups.data.should.have.length(2);
            data.hostGroups.meta.count.should.equal(2);
            data.hostGroups.meta.total.should.equal(5);
            expect(data).toMatchSnapshot();
        });

        test('limit + offset', async () => {
            const {data, status} = await runQuery(GROUP_FILTERS_QUERY, {
                limit: 1,
                offset: 1
            }, headers);

            data.hostGroups.data.should.have.length(1);
            data.hostGroups.meta.should.have.property('count', 1);
            data.hostGroups.meta.should.have.property('total', 5);

            expect(status).toEqual(200);
            expect(data).toMatchSnapshot();
        });

        test('ordering', async () => {
            const {data, status} = await runQuery(GROUP_FILTERS_QUERY, {
                order_by: 'group',
                order_how: 'ASC'
            });

            expect(status).toEqual(200);
            expect(data).toMatchSnapshot();
        });
    });

    describe('host filters', function () {
        test('by id', async () => {
            const { data, status } = await runQuery(GROUP_FILTERS_QUERY, {
                hostFilter: { id: { eq: 'd36bd5c3-fb39-4826-8d82-152bedfd1ecd' }}
            });

            expect(status).toEqual(200);
            expect(data).toMatchSnapshot();
        });
    });
});

