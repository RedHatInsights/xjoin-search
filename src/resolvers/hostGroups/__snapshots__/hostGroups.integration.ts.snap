// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`host groups basic query 1`] = `
Object {
  "hostGroups": Object {
    "data": Array [
      Object {
        "count": 1,
        "group": Object {
          "account": "test",
          "created_on": "2023-04-12T11:11:00.000000Z",
          "id": "22cd8e39-13bb-4d02-8316-84b850dc5136",
          "modified_on": "2023-04-12T12:34:56.789000Z",
          "name": "group1",
          "org_id": "test",
        },
      },
    ],
    "meta": Object {
      "count": 1,
      "total": 1,
    },
  },
}
`;

exports[`host groups host filters by id 1`] = `
Object {
  "hostGroups": Object {
    "data": Array [],
    "meta": Object {
      "count": 0,
      "total": 0,
    },
  },
}
`;

exports[`host groups pagination limit + offset 1`] = `
Object {
  "hostGroups": Object {
    "data": Array [
      Object {
        "count": 1,
        "group": Object {
          "account": "hostGroupsTest",
          "created_on": "2023-04-12T11:11:00.000000Z",
          "id": "7c004532-13ff-4741-ab19-53bcb610490e",
          "modified_on": "2023-04-12T12:34:56.789000Z",
          "name": "baz",
          "org_id": "hostGroupsTest",
        },
      },
    ],
    "meta": Object {
      "count": 1,
      "total": 6,
    },
  },
}
`;

exports[`host groups pagination limit 1`] = `
Object {
  "hostGroups": Object {
    "data": Array [
      Object {
        "count": 2,
        "group": Object {
          "account": "hostGroupsTest",
          "created_on": "2023-04-12T11:11:00.000000Z",
          "id": "54ef7a4d-47fd-47f9-85e3-d60259e2d8f2",
          "modified_on": "2023-04-12T12:34:56.789000Z",
          "name": "prod",
          "org_id": "hostGroupsTest",
        },
      },
      Object {
        "count": 1,
        "group": Object {
          "account": "hostGroupsTest",
          "created_on": "2023-04-12T11:11:00.000000Z",
          "id": "7c004532-13ff-4741-ab19-53bcb610490e",
          "modified_on": "2023-04-12T12:34:56.789000Z",
          "name": "baz",
          "org_id": "hostGroupsTest",
        },
      },
    ],
    "meta": Object {
      "count": 2,
      "total": 6,
    },
  },
}
`;

exports[`host groups pagination ordering 1`] = `
Object {
  "hostGroups": Object {
    "data": Array [
      Object {
        "count": 1,
        "group": Object {
          "account": "test",
          "created_on": "2023-04-12T11:11:00.000000Z",
          "id": "22cd8e39-13bb-4d02-8316-84b850dc5136",
          "modified_on": "2023-04-12T12:34:56.789000Z",
          "name": "group1",
          "org_id": "test",
        },
      },
    ],
    "meta": Object {
      "count": 1,
      "total": 1,
    },
  },
}
`;
